File: ABC/ABC027/B/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);

    int N = sc.nextInt();
    int[] a = new int[N];
    int sum = 0;

    for (int i = 0; i < N; i++) {
      a[i] = sc.nextInt();
      sum += a[i];
    }

    sc.close();

    if (sum % N != 0) {
      System.out.println(-1);
    } else {

      int count = 0;
      
      int average = sum / N;

      int current = 0;

      for (int i = 0; i < N; i++) {
        
        current += a[i];

        if ((i + 1) * average != current) {
          count++;
        }

      }

      System.out.println(count);
      
    }
  }

}

File: ABC/ABC027/B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int N;
  cin >> N;

  int sum = 0;
  vector<int> a(N);

  for (int i = 0; i < N; i++) {
    cin >> a[i];
    sum += a[i];
  }

  int average = sum / N;
  int current = 0;
  int answer = 0;

  if (sum % N != 0) {
    answer = -1;
  } else {
    for (int i = 0; i < N; i++) {
      current += a[i];
      if ((i + 1) * average != current) {
        answer++;
      }
    }
  }

  cout << answer << endl;
 
}